spatialmath/geo_obstacle.go

```diff
@@ -110,3 +106,35 @@ func GeoObstaclesFromConfig(config *GeoObstacleConfig) ([]*GeoObstacle, error) {
 	}
 	return gobs, nil
 }
+
+// GetCartesianDistance calculates the great circle distance between p and q.
+func GetCartesianDistance(p, q *geo.Point) (float64, float64) {
+	mod := geo.NewPoint(p.Lat(), q.Lng())
+	// Calculates the Haversine distance between two points in kilometers
+	latDist := p.GreatCircleDistance(mod)
+	lngDist := q.GreatCircleDistance(mod)
+	return latDist, lngDist
+}
+
+// GeoPointToPose converts p into a spatialmath pose relative to lng = 0 = lat.
+func GeoPointToPose(p *geo.Point) Pose {
+	latDist, lngDist := GetCartesianDistance(geo.NewPoint(0, 0), p)
+	// multiple by 1000000 to convert km to mm
+	return NewPoseFromPoint(r3.Vector{latDist * 1000000, lngDist * 1000000, 0})
```

Comment:
I think for GoToInputs it will be problematic that this doesn't include an orientation, since it will try to spin until it gets the desired orientation. Can we expect a compass to be part of the movement sensor?

Reply:
yea, we can assert that the localizer is of type movementSensor and then use the `CompassHeading` or `Orientation` method.